src/TwigComponent/src/ComponentPropertiesExtractor.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Twig\PropsNode;
+use Twig\Environment;
+
+/**
+ * @author Jean-François Lépine <lepinejeanfrancois@gmail.com>
+ */
+final class ComponentPropertiesExtractor
+{
+    public function __construct(
+        private readonly Environment $twig,
+    ) {
+    }
+
+    /**
+     * @return array<array{display: string, name: string, type: string, default: mixed}>
```

Comment:
I recommended you to add a description, and also a warning like it has been done on `\Symfony\Component\Routing\RouterInterface::getRouteCollection()`: 

```suggestion
     * Returns a list of properties from a Component. 
     *
     * Warning: We do not recommend using this method at runtime, as it is rather slow.
```

Reply:
I don't think this service should be available when not in debug for now.